### PR TITLE
Rebalance unit test and integration test time for github action

### PR DESCRIPTION
**Pull request by xiangfu0**
Try to rebalance the test to reduce the overall time for CI jobs.

### CODE REVIEW

**Review by gortiz — APPROVED** [review]

**Comment by xiangfu0** [comment]
Current setup seems pretty balanced on the runtime for unit test and integration tests runtime.
https://github.com/apache/pinot/actions/runs/5730948007/job/15530794730?pr=11230

File: .github/workflows/scripts/.pinot_test.sh [hunk 1]
```diff
@@ -56,18 +56,11 @@ else
         -pl 'pinot-core' \
         -pl 'pinot-query-planner' \
         -pl 'pinot-query-runtime' \
-        -pl 'pinot-server' \
```
Comment (walterddr) [inline .github/workflows/scripts/.pinot_test.sh:59]: Let's wait and check the final build summary and check there's no overlap

**Review by walterddr — APPROVED** [review]
LGTM

File: .github/workflows/scripts/.pinot_test.sh (hunk 1 above)
Comment (xiangfu0) [reply .github/workflows/scripts/.pinot_test.sh:59]: This is for unit set 1:
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Pinot 0.13.0-SNAPSHOT:
[INFO] 
[INFO] Pinot .............................................. SUCCESS [  4.902 s]
[INFO] Pinot Service Provider Interface ................... SUCCESS [ 15.603 s]
[INFO] Pinot Segment Service Provider Interface ........... SUCCESS [ 39.404 s]
[INFO] Pinot Plugins ...................................... SUCCESS [  0.837 s]
[INFO] Pinot Metrics ...................................... SUCCESS [  0.499 s]
[INFO] Pinot Yammer Metrics ............................... SUCCESS [  9.953 s]
[INFO] Pinot FMPP plugin .................................. SUCCESS [  3.299 s]
[INFO] Pinot Common ....................................... SUCCESS [02:03 min]
[INFO] Pinot Input Format ................................. SUCCESS [  0.583 s]
[INFO] Pinot Avro Base .................................... SUCCESS [ 15.172 s]
[INFO] Pinot Avro ......................................... SUCCESS [  2.863 s]
[INFO] Pinot Csv .......................................... SUCCESS [ 15.870 s]
[INFO] Pinot JSON ......................................... SUCCESS [ 14.520 s]
[INFO] Pinot local segment implementations ................ SUCCESS [11:41 min]
[INFO] Pinot Core ......................................... SUCCESS [23:42 min]
[INFO] Pinot Query Planner ................................ SUCCESS [ 51.010 s]
[INFO] Pinot Query Runtime ................................ SUCCESS [02:40 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  43:03 min
[INFO] Finished at: 2023-08-01T21:38:43Z
[INFO] ------------------------------------------------------------------------
```

Comment (xiangfu0) [reply .github/workflows/scripts/.pinot_test.sh:59]: This is for test set 2:
```
2023-08-01T21:41:53.0116496Z [INFO] ------------------------------------------------------------------------
2023-08-01T21:41:53.0117964Z [INFO] Reactor Summary for Pinot 0.13.0-SNAPSHOT:
2023-08-01T21:41:53.0118577Z [INFO] 
2023-08-01T21:41:53.0120466Z [INFO] Pinot .............................................. SUCCESS [  4.819 s]
2023-08-01T21:41:53.0121145Z [INFO] Pinot Plugins ...................................... SUCCESS [  1.199 s]
2023-08-01T21:41:53.0122432Z [INFO] Pinot Metrics ...................................... SUCCESS [  0.215 s]
2023-08-01T21:41:53.0123789Z [INFO] Pinot Yammer Metrics ............................... SUCCESS [  3.927 s]
2023-08-01T21:41:53.0124711Z [INFO] Pinot FMPP plugin .................................. SUCCESS [  2.403 s]
2023-08-01T21:41:53.0125098Z [INFO] Pinot Input Format ................................. SUCCESS [  0.350 s]
2023-08-01T21:41:53.0125678Z [INFO] Pinot Avro Base .................................... SUCCESS [ 13.520 s]
2023-08-01T21:41:53.0126056Z [INFO] Pinot Avro ......................................... SUCCESS [  0.461 s]
2023-08-01T21:41:53.0126757Z [INFO] Pinot Csv .......................................... SUCCESS [ 17.071 s]
2023-08-01T21:41:53.0127265Z [INFO] Pinot JSON ......................................... SUCCESS [ 14.939 s]
2023-08-01T21:41:53.0127619Z [INFO] Pinot Server ....................................... SUCCESS [01:22 min]
2023-08-01T21:41:53.0127970Z [INFO] Pinot Segment Uploader ............................. SUCCESS [  1.928 s]
2023-08-01T21:41:53.0128328Z [INFO] Pinot Segment Uploader Default ..................... SUCCESS [  1.868 s]
2023-08-01T21:41:53.0128685Z [INFO] Pinot Controller ................................... SUCCESS [11:26 min]
2023-08-01T21:41:53.0129211Z [INFO] Pinot Broker ....................................... SUCCESS [02:28 min]
2023-08-01T21:41:53.0129706Z [INFO] Pinot Clients ...................................... SUCCESS [  0.468 s]
2023-08-01T21:41:53.0130039Z [INFO] Pinot Java Client .................................. SUCCESS [  9.590 s]
2023-08-01T21:41:53.0130708Z [INFO] Pinot JDBC Client .................................. SUCCESS [  4.070 s]
2023-08-01T21:41:53.0131453Z [INFO] Pinot Batch Ingestion .............................. SUCCESS [  1.378 s]
2023-08-01T21:41:53.0131820Z [INFO] Pinot Batch Ingestion Common ....................... SUCCESS [  1.825 s]
2023-08-01T21:41:53.0132219Z [INFO] Pinot Minion ....................................... SUCCESS [  9.913 s]
2023-08-01T21:41:53.0132982Z [INFO] Pinot CLP Log ...................................... SUCCESS [  2.397 s]
2023-08-01T21:41:53.0133353Z [INFO] Pinot Confluent Avro ............................... SUCCESS [  0.585 s]
2023-08-01T21:41:53.0135469Z [INFO] Pinot ORC .......................................... SUCCESS [ 11.824 s]
2023-08-01T21:41:53.0137271Z [INFO] Pinot Parquet ...................................... SUCCESS [01:13 min]
2023-08-01T21:41:53.0141209Z [INFO] Pinot Thrift ....................................... SUCCESS [  3.556 s]
2023-08-01T21:41:53.0143890Z [INFO] Pinot Protocol Buffers ............................. SUCCESS [01:29 min]
2023-08-01T21:41:53.0146940Z [INFO] Pluggable Pinot file system ........................ SUCCESS [  0.251 s]
2023-08-01T21:41:53.0150044Z [INFO] Pinot Azure Data Lake Storage ...................... SUCCESS [  9.708 s]
2023-08-01T21:41:53.0152647Z [INFO] Pinot Hadoop Filesystem ............................ SUCCESS [  2.330 s]
2023-08-01T21:41:53.0157397Z [INFO] Pinot Google Cloud Storage ......................... SUCCESS [  2.179 s]
2023-08-01T21:41:53.0157793Z [INFO] Pinot Amazon S3 .................................... SUCCESS [ 22.136 s]
2023-08-01T21:41:53.0158162Z [INFO] Pinot Batch Ingestion for Spark 2.4 ................ SUCCESS [  7.036 s]
2023-08-01T21:41:53.0158700Z [INFO] Pinot Batch Ingestion for Spark 3.2 ................ SUCCESS [  7.520 s]
2023-08-01T21:41:53.0159097Z [INFO] Pinot Batch Ingestion for Hadoop ................... SUCCESS [  8.892 s]
2023-08-01T21:41:53.0159490Z [INFO] Pinot Batch Ingestion Standalone ................... SUCCESS [  3.573 s]
2023-08-01T21:41:53.0159872Z [INFO] Pinot Stream Ingestion ............................. SUCCESS [  0.259 s]
2023-08-01T21:41:53.0160443Z [INFO] Pinot Kafka Base ................................... SUCCESS [  2.725 s]
2023-08-01T21:41:53.0160808Z [INFO] Pinot Kafka 2.x .................................... SUCCESS [01:26 min]
2023-08-01T21:41:53.0161166Z [INFO] Pinot Kinesis ...................................... SUCCESS [  3.968 s]
2023-08-01T21:41:53.0161531Z [INFO] Pinot Pulsar ....................................... SUCCESS [01:28 min]
2023-08-01T21:41:53.0161898Z [INFO] Pinot Minion Tasks ................................. SUCCESS [  4.510 s]
2023-08-01T21:41:53.0162495Z [INFO] Pinot Minion Built-In Tasks ........................ SUCCESS [ 53.187 s]
2023-08-01T21:41:53.0162865Z [INFO] Pinot Dropwizard Metrics ........................... SUCCESS [  2.732 s]
2023-08-01T21:41:53.0163242Z [INFO] Pinot Segment Writer ............................... SUCCESS [  1.158 s]
2023-08-01T21:41:53.0163624Z [INFO] Pinot Segment Writer File Based .................... SUCCESS [  6.423 s]
2023-08-01T21:41:53.0164084Z [INFO] Pluggable Pinot Environment Provider ............... SUCCESS [  0.157 s]
2023-08-01T21:41:53.0164471Z [INFO] Pinot Azure Environment ............................ SUCCESS [  3.732 s]
2023-08-01T21:41:53.0164838Z [INFO] Pinot Tools ........................................ SUCCESS [ 22.625 s]
2023-08-01T21:41:53.0170710Z [INFO] Pinot Test Utils ................................... SUCCESS [ 11.300 s]
2023-08-01T21:41:53.0171348Z [INFO] Pinot Integration Tests ............................ SUCCESS [  8.771 s]
2023-08-01T21:41:53.0171738Z [INFO] Pinot Perf ......................................... SUCCESS [  7.491 s]
2023-08-01T21:41:53.0172099Z [INFO] Pinot Distribution ................................. SUCCESS [  5.283 s]
2023-08-01T21:41:53.0172456Z [INFO] Pinot Connectors ................................... SUCCESS [  0.526 s]
2023-08-01T21:41:53.0172821Z [INFO] Pinot Spark Common ................................. SUCCESS [  4.687 s]
2023-08-01T21:41:53.0173197Z [INFO] Pinot Spark 2 Connector ............................ SUCCESS [  5.196 s]
2023-08-01T21:41:53.0173572Z [INFO] Pinot Spark 3 Connector ............................ SUCCESS [  5.763 s]
2023-08-01T21:41:53.0173934Z [INFO] Pinot Flink Connector .............................. SUCCESS [ 30.583 s]
2023-08-01T21:41:53.0174509Z [INFO] Pinot Compatibility Verifier ....................... SUCCESS [  6.830 s]
2023-08-01T21:41:53.0175100Z [INFO] ------------------------------------------------------------------------
2023-08-01T21:41:53.0175435Z [INFO] BUILD SUCCESS
2023-08-01T21:41:53.0175870Z [INFO] ------------------------------------------------------------------------
2023-08-01T21:41:53.0176210Z [INFO] Total time:  26:57 min
2023-08-01T21:41:53.0179039Z [INFO] Finished at: 2023-08-01T21:41:53Z
2023-08-01T21:41:53.0179505Z [INFO] ------------------------------------------------------------------------
```

Comment (xiangfu0) [reply .github/workflows/scripts/.pinot_test.sh:59]: Basically no overlap.

**Comment by xiangfu0** [comment]
Another run: https://github.com/apache/pinot/actions/runs/5730948007?pr=11230
Better balanced than before.

**Pull request merged**